x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -4,58 +4,82 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { i18n } from '@kbn/i18n';
+import { curry } from 'lodash';
 import axios, { AxiosError, AxiosResponse } from 'axios';
 import { schema, TypeOf } from '@kbn/config-schema';
 import { getRetryAfterIntervalFromHeaders } from './lib/http_rersponse_retry_header';
 import { nullableType } from './lib/nullable';
 import { isOk, promiseResult, Result } from './lib/result_type';
 import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { ActionsConfigurationUtilities } from '../actions_config';
 
 // config definition
 enum WebhookMethods {
   POST = 'post',
   PUT = 'put',
 }
 
-export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
-
 const HeadersSchema = schema.recordOf(schema.string(), schema.string());
-
-const ConfigSchema = schema.object({
+const configSchemaProps = {
   url: schema.string(),
   method: schema.oneOf([schema.literal(WebhookMethods.POST), schema.literal(WebhookMethods.PUT)], {
     defaultValue: WebhookMethods.POST,
   }),
   headers: nullableType(HeadersSchema),
-});
+};
+const ConfigSchema = schema.object(configSchemaProps);
+type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
 
 // secrets definition
-export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
 const SecretsSchema = schema.object({
   user: schema.string(),
   password: schema.string(),
 });
 
 // params definition
-export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+type ActionParamsType = TypeOf<typeof ParamsSchema>;
 const ParamsSchema = schema.object({
   body: schema.maybe(schema.string()),
 });
 
 // action type definition
-export const actionType: ActionType = {
-  id: '.webhook',
-  name: 'webhook',
-  validate: {
-    config: ConfigSchema,
-    secrets: SecretsSchema,
-    params: ParamsSchema,
-  },
-  executor,
-};
+export function getActionType(configurationUtilities: ActionsConfigurationUtilities): ActionType {
+  return {
+    id: '.webhook',
+    name: 'webhook',
+    validate: {
+      config: schema.object(configSchemaProps, {
+        validate: curry(valdiateActionTypeConfig)(configurationUtilities),
+      }),
+      secrets: SecretsSchema,
+      params: ParamsSchema,
+    },
+    executor: curry(executor)(configurationUtilities),
+  };
+}
+
+function valdiateActionTypeConfig(
+  configurationUtilities: ActionsConfigurationUtilities,
+  configObject: ActionTypeConfigType
+) {
+  try {
+    configurationUtilities.ensureWhitelistedUri(configObject.url);
+  } catch (whitelistError) {
+    return i18n.translate('xpack.actions.builtin.webhook.webhookConfigurationError', {
+      defaultMessage: 'error configuring webhook action: {message}',
```

Comment:
I think we still need the actionId in here - unless it's somehow included in some other way (not obvious to me that it is).

If we need to get it added, can be done in a subsequent PR; create an issue | project card for it.

Reply:
Yeah, we don't have access to the ID at this point (in create it happens before an ID is generated, and in update/execute it's not provided), so this will have to be a separate PR.